drivers/console/Kconfig

```
@@ -146,6 +146,38 @@ config RTT_TX_RETRY_IN_INTERRUPT
 
 endif
 
+config SEMIHOSTING_CONSOLE
+	bool "Use semihosting console"
+	depends on CPU_CORTEX_M
+	select CONSOLE_HAS_DRIVER
+	help
+	  Enable this option to use semihosting for console.
+	  Semihosting is a mechanism that enables code running on an ARM target
+	  to communicate and use the Input/Output facilities on a host computer
+	  that is running a debugger.
+	  Additional information can be found in:
+	  https://developer.arm.com/docs/dui0471/k/what-is-semihosting/what-is-semihosting
+	  This option is compatible with hardware and with QEMU, through the
+	  (automatic) use of the -semihosting-config switch when invoking it.
+
+if SEMIHOSTING_CONSOLE
+
+config SEMIHOSTING_BUFFER_SIZE
+	int "Buffer size"
+	default 128
+	range 1 255
+	help
+	  Size of the semihosting console buffer in bytes
+
+config SEMIHOSTING_SEND_TIMEOUT_MS
```

Since we are using the term 'flush' in the code, it would be nice to call the related config flush as well.
```suggestion
config SEMIHOSTING_CONSOLE_FLUSH_TIMEOUT
```